include/solvers/navier_stokes_assemblers.h

```diff
@@ -202,10 +202,14 @@ class GLSNavierStokesAssemblerNonNewtonianCore
                          const double &        non_newtonian_viscosity,
                          const double &        d_gamma_dot) const
   {
+    std::map<field, double> field_values_plus;
+
     // Calculates an approximation of the derivative of the viscosity with a
     // slight change in the shear rate magnitude using finite difference
+    field_values_plus[field::shear_rate] = shear_rate_magnitude + d_gamma_dot;
+
     const double non_newtonian_viscosity_plus =
-      rheological_model->get_viscosity(shear_rate_magnitude + d_gamma_dot);
+      rheological_model->value(field_values_plus);
```

Comment:
I don't really have a suggestion for this, but unlike the other classes where the name of the classe is the name of the output value, here it's a bit weird that rheological_model output a viscosity without specification. But I guess this is the best way to do it anyways.

Reply:
Well in a way a rheology model should always output a viscosity. so I don't think that's an issue.